### PR TITLE
DV2: Fix column name collision detection

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/oncall/issues/3037

* canonical name should be up/downcase of the escaped name
* add tests
    * ... maybe this should be in the sqlgenerator integration test class?
    * ....... maybe we should find a real contract-testing library

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`6c5538f6e4`](https://github.com/airbytehq/airbyte/pull/30739/commits/6c5538f6e4e72f726a7f3e9d18772b16259828a6)) - ❌

⏲️  Total pipeline duration: 10mn06s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Build connector tar                                                          | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                    | ❌        |
| Java Connector Integration Tests                                             | ❌        |
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| Connector version increment check                                            | ✅        |
| QA checks                                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695667108/6c5538f6e4e72f726a7f3e9d18772b16259828a6/destination-snowflake/3.1.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6c5538f6e4e72f726a7f3e9d18772b16259828a6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`6c5538f6e4`](https://github.com/airbytehq/airbyte/pull/30739/commits/6c5538f6e4e72f726a7f3e9d18772b16259828a6)) - ✅

⏲️  Total pipeline duration: 07mn09s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695667108/6c5538f6e4e72f726a7f3e9d18772b16259828a6/destination-bigquery/2.0.17/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6c5538f6e4e72f726a7f3e9d18772b16259828a6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java [hunk 1]
```diff
@@ -60,11 +59,8 @@ public StreamId buildStreamId(final String namespace, final String name, final S
 
   @Override
   public ColumnId buildColumnId(final String name, final String suffix) {
-    // No escaping needed, as far as I can tell. We quote all our identifier names.
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java:63]: :p

**Review by evantahler — APPROVED** [review]
👍 and good tests!

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`926ec50cc5`](https://github.com/airbytehq/airbyte/pull/30739/commits/926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2)) - ✅

⏲️  Total pipeline duration: 10mn49s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695848497/926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2/destination-bigquery/2.0.18/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`926ec50cc5`](https://github.com/airbytehq/airbyte/pull/30739/commits/926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2)) - ❌

⏲️  Total pipeline duration: 09mn01s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Build connector tar                                                          | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                    | ✅        |
| Java Connector Integration Tests                                             | ❌        |
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| Connector version increment check                                            | ✅        |
| QA checks                                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695848497/926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2/destination-snowflake/3.1.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:926ec50cc5b4c0a1cfe6d398434879c6ea54fdc2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`03cd0d7b0a`](https://github.com/airbytehq/airbyte/pull/30739/commits/03cd0d7b0ad476c182db5f89e15a942006308e2e)) - ✅

⏲️  Total pipeline duration: 09mn38s 

| Step                                                                         | Result   |
|:-----------------------------------------------------------------------------|:---------|
| Build connector tar                                                          | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                    | ✅        |
| Java Connector Integration Tests                                             | ✅        |
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml | ✅        |
| Connector version semver check                                               | ✅        |
| Connector version increment check                                            | ✅        |
| QA checks                                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695852477/03cd0d7b0ad476c182db5f89e15a942006308e2e/destination-snowflake/3.1.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:03cd0d7b0ad476c182db5f89e15a942006308e2e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`03cd0d7b0a`](https://github.com/airbytehq/airbyte/pull/30739/commits/03cd0d7b0ad476c182db5f89e15a942006308e2e)) - ✅

⏲️  Total pipeline duration: 03mn33s 

| Step                                                                        | Result   |
|:----------------------------------------------------------------------------|:---------|
| Build connector tar                                                         | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                   | ✅        |
| Java Connector Integration Tests                                            | ✅        |
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml | ✅        |
| Connector version semver check                                              | ✅        |
| Connector version increment check                                           | ✅        |
| QA checks                                                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_column_collisions/1695852477/03cd0d7b0ad476c182db5f89e15a942006308e2e/destination-bigquery/2.0.18/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:03cd0d7b0ad476c182db5f89e15a942006308e2e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Pull request merged**